app/stores/toplevel/ProfileStore.js

```diff
@@ -93,6 +128,32 @@ export default class SettingsStore extends Store {
     await this.getProfileLocaleRequest.execute(); // eagerly cache
   };
 
+  _updateTheme = async ({ theme }: { theme: string }) => {
+    await this.unsetCustomThemeRequest.execute();
+    await this.getCustomThemeRequest.execute(); // eagerly cache
+    await this.setThemeRequest.execute(theme);
+    await this.getThemeRequest.execute(); // eagerly cache
+  };
+
+  _exportTheme = async () => {
+    // TODO: It should be ok to access DOM Style from here
+    // but not sure about project conventions about accessing the DOM (Clark)
+    const html = document.querySelector('html');
+    if (html) {
+      const attributes: any = html.attributes;
+      await this.unsetCustomThemeRequest.execute();
+      await this.setCustomThemeRequest.execute(attributes.style.value);
+      await this.getCustomThemeRequest.execute(); // eagerly cache
+    }
+  };
+
+
+  // TODO: not sure where to put this method (Clark)
+  getThemeVars = ({ theme }: { theme: string }) => {
```

Comment:
I think this function is doing the wrong thing. I should take your currently selected theme and merge in any custom overrides into it.

Reply:
Hi this is a helper function that basically takes a `theme` name (one of the pre-built themes) and extract the CSS variables of the specific theme.

This is helpful specially when trying to generate the preview of the thumbnails because I can access the variables any theme.

So if I was trying to generate a preview of `YoroiModern` theme, I will utilize this function to get the `YoroiModern` variables and then dynamically change properties of the SVG for example based on those variables.

Please let me know what you think/criticism or maybe if I have misunderstood what you meant.

Thank You!

Reply:
Ah my bad I didn't notice this function was used in multiple places. That means the merge of the selected theme & custom theme should happen in `currentThemeVars`, correct?

Reply:
Yep! I added a new commit to improve and make this behavior more obvious as-well as optimized